R/return_linter.R

```diff
@@ -0,0 +1,73 @@
+#' Return linter
+#'
+#' This linter checks for explicit [return()] at the end of a function
+#'
+#' @param use_implicit_returns Whether to use implicit or explicit returns
+#'
+#' @evalRd rd_tags("return_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#'
+#' @export
+return_linter <- function(use_implicit_returns = TRUE) {
+
+  if (use_implicit_returns) {
+    xpath <- "
+      (//FUNCTION | //OP-LAMBDA)[following-sibling::expr[1]/*[1][self::OP-LEFT-BRACE]]
```

Comment:
we should be able to combine the repetitive `following-sibling::expr[1]`, e.g.

```
(//FUNCTION | //OP-LAMBDA)
  /following-sibling::expr[1][OP-LEFT-BRACE]
  /expr[last()][...]
```

ditto for the XPath in the other branch

Reply:
If you leave that out, wouldn't you have a problem with in-line functions? This way you can easily exclude them from the lint

Reply:
See updated `use_implicit_returns` XPath